### ISSUE TITLE
[API] Should FlexItem props be added to COMMON props?

### ISSUE THREAD

**Issue by dmarcey** [primer/components#701]
Today the only component that takes flex item props (props related to being the child of a flexbox like `flex-grow` or `flex-shrink`) is `<Flex />` itself. This prevents someone from doing something like this (full example: https://codesandbox.io/s/affectionate-jones-rsl84):

```
<Flex flexDirection="row">
  <Box flexGrow={1} bg="blue.2">
    Box with flex-grow:1
  </Box>
  <Box flexShrink={0} bg="green.2">
    Box with flex-shrink:0
  </Box>
</Flex>
```

And it's not just `<Box>`'s where someone might want some flex item styles, I think that pretty much any component could be a child of a flexbox, so should we add the props associated with being a flex item to our `COMMON` styled-system `styleFn`?

```
export const COMMON = compose(styledSystem.space, styledSystem.color, styledSystem.display)
COMMON.propTypes = {
  ...systemPropTypes.space,
  ...systemPropTypes.color
}
```

I think it would be the props under the //item comment here:
https://github.com/styled-system/styled-system/blob/master/packages/flexbox/src/index.js
```
flex: true,
flexGrow: true,
flexShrink: true,
flexBasis: true,
justifySelf: true,
alignSelf: true,
order: true,
```

If you look at the API for flexbox in `styled-system` (https://styled-system.com/api#flexbox) it shows examples where like `<Box flex='1 1 auto' />`, so I don't think we would be out of line to do something similar.

**Comment by emplums** [comment]
@dmarcey The `Flex` component has all the same props as the `Box` component (it's based off of `Box`) - what would be the advantage to using `Box` here instead of `Flex` for the child?

**Comment by dmarcey** [comment]
Sometimes you might not want an element with  with `display: flex`, just to get the "flex item" props.

I definitely get the limited surface area balance. It looks like on the styled-system docs they expect a `<Box />` to be able to take a `flex` prop. Maybe it's worth looking at other libraries that use `styled-system` to see what they do in this case?

**Comment by emplums** [comment]
@dmarcey that makes sense! In that case you'd have to reset the `display` prop otherwise you might have some side effects that you don't expect by making the child component `display: flex` which doesn't feel good!

I can do some research on other libraries this week!

**Comment by emplums** [comment]
For some historical context, we used to have a separate `Flex.Item` component that had all the flex child properties + flex parent properties (in case you wanted to have the child also be a flex parent). I think we ended up consolidating it all into one component for ease of use, but I can see why that can be troublesome to have `display: flex` set by default. Might be worth considering adding `Flex.Item` back and not having the display set by default.
